ssg/build_yaml.py

```diff
@@ -645,107 +677,166 @@ def to_file(self, file_name):
         tree.write(file_name)
 
 
-def load_benchmark_or_group(group_file, benchmark_file, guide_directory, action,
-                            profiles_dir, env_yaml, bash_remediation_fns):
-    """
-    Loads a given benchmark or group from the specified benchmark_file or
-    group_file, in the context of guide_directory, action, profiles_dir,
-    env_yaml, and bash_remediation_fns.
+class DirectoryLoader(object):
+    def __init__(self, profiles_dir, bash_remediation_fns, env_yaml):
+        self.benchmark_file = None
+        self.group_file = None
+        self.loaded_group = None
+        self.rules = []
+        self.values = []
+        self.subdirectories = []
+
+        self.profiles_dir = profiles_dir
+        self.bash_remediation_fns = bash_remediation_fns
+        self.env_yaml = env_yaml
+
+        self.parent_group = None
+
+    def _examine_directory(self, guide_directory):
```

Comment:
What is the directory being examined for?
I would say pre-loading.

Reply:
preloading is also abstract and, in a way, fuzzy - but what about collecting?

Reply:
Collect seems good.